### PR TITLE
Do not show a window if there are no qtgui blocks.

### PR DESCRIPTION
**Pull request by krk**
Fixes https://github.com/gnuradio/gnuradio/issues/2021

### CODE REVIEW

File: grc/core/generator/top_block.py [hunk 1]
```diff
@@ -111,14 +112,17 @@ def _build_python_code_from_template(self):
             elif key == 'epy_module':
                 src = block.params['source_code'].get_value()
                 output.append((file_path, src))
+            elif not hasQtGui and "qtgui" in key:
```
Comment (marcusmueller) [inline grc/core/generator/top_block.py]: this is a bit redundant: the first condition `not hasQtGui` could be omitted.

Anyway, this is a bit *too* shallow for actually checking for whether a QtGui element exists in a flow graph. We're not forcing anyone to use "qtgui" in the names of their graphical sinks, and there's multiple out-of-tree visualization sinks that don't contain that string.

Comment (krk) [reply grc/core/generator/top_block.py]: That makes sense, I have only checked the in-tree blocks to see if they have something in common and that is how I arrived at name having "qtgui".

What about checking the python import statements to see if they contain `from PyQt5 import Qt`?

Comment (marcusmueller) [reply grc/core/generator/top_block.py]: Hm, that's a very hard question. Maybe we should be forcing people to explicitly declare their blocks as having QT windows, simply by having a new `flag` for the YAML files. That would make your life a lot easier here, wouldn't it?

**Comment by mbr0wn** [comment]
I would love this behaviour. All of @marcusmueller 's points are valid too though.

**Comment by nickoe** [comment]
@krk, do you intend to follow up on this pull request?

**Comment by krk** [comment]
@nickoe Thank you for the ping. I have added a `hide_qt_window` option instead and rebased on the latest master.

File: grc/core/generator/top_block.py (hunk 1 above)
Comment (sdh11) [reply grc/core/generator/top_block.py]: Qt blocks get added to the `top_grid_layout` in the flowgraph object in order to show up in the window. We can check the `count` of the layout to see if any blocks have been added to it and if not, hide the window. Unless someone is manually modifying the generated python, that should have all the included blocks.

Comment (sdh11) [reply grc/core/generator/top_block.py]: Having the explicit option to override and hide the window even if blocks exist would be nice too.

**Comment by nickoe** [comment]
@krk what do we gain from having this hide option when you could just select the "No GUI" option?

**Comment by mormj** [comment]
I agree with @nickoe here - the real power in this feature will be to not open a window when there are no blocks. If there is a manual step involved, you may as well just select the no gui option.

**Comment by nickoe** [comment]
@krk This pull request is becoming stale. It looks to me like you have misunderstood wish in #2021. The idea is to have it it _automatically_ detect if ti should how a window or now depending on any gui blocks being present and active.

**Comment by mbr0wn** [comment]
Looks like the consensus here is that this is a suboptimal solution. @krk thanks for your contribution, hopefully we can see more!

**Pull request closed without merging**